sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -542,6 +542,11 @@ case class DataSource(
       throw new AnalysisException("Cannot save interval data type into external storage.")
     }
 
+    if (data.schema.size == 0) {
```

Comment:
Is it required? This is a behavior change. Can we exclude it from this PR?

Reply:
@gatorsmile May i request you to please quickly go through Wenchen's and Ryan's comments  above ? My understanding is that , we want to consistently rejecting writing empty schema for all the data sources ? Please let me know.

Reply:
Currently, we are not blocking this. I do not think we should introduce this behavior change. This is risky to block all the cases. 

Previously, I tried to block CREATE TABLE with an empty schema. Later, I hit a regression because some data sources are using options/table properties to specify the schema...

A general guide here is to avoid behavior changes if possible. When we have to introduce a behavior change, we should make it configurable. At least, users can convert it back by using a flag.

Reply:
@gatorsmile @cloud-fan OK.. sounds reasonable to me. I will rollback the latest change in this PR and we can discuss if we want to introduce the behaviour change in a future jira/pr.